### ISSUE TITLE
Packet Corruption Problem?

### ISSUE THREAD

**Issue by mhkndo** [pmacct/pmacct#486]
Hello all,

I've a project that I'd like to get hostname from client hello messages in TLS protocol. For this I've found that ndpi can be used to get the related data out of client hello message we received. However there is a packet corruption problem I'm facing currently and I'm also not so sure if it is a corruption or I did something wrong. 

For every packet we received in "pm_pcap_cb" callback function I'm printing whole packet to stdout and I'm comparing it with the one I see on wireshark. Interestingly for the same packet every byte is same until "cipher suite len". And because of this cipher suite len is read wrong in the ndpi library and therefore remaining parts are also parsed wrong.

To show the problem here is the "client hello" packet that I printed to stdout in "pm_pcap_cb" callback function:
[stdout.txt](https://github.com/pmacct/pmacct/files/6211614/stdout.txt)

hex dump of the same packet from wireshark:
[wireshark.txt](https://github.com/pmacct/pmacct/files/6211620/wireshark.txt)

In summary the packet I received in pm_pcap_cb is different with wireshark
So any ideas?

**Comment by mhkndo** [comment]
I've found the problem. Default snaplen is set to 128 in the beginning of main function in pmacctd.c. Because of this payload of the packets are not read completely. I guess for those who wants to work with ndpi needs a bigger snaplen to get all fields of packets.

I'm closing this issue.
Best Regards
mhkndo

**Comment by paololucente** [comment]
Hi @mhkndo , thank you very much for confirming. Looking at snaplen would have indeed been my first suggestion. Paolo
